pkg/oauthserver/server/csrf/cookie.go

```diff
@@ -28,23 +28,23 @@ func NewCookieCSRF(name, path, domain string, secure, httponly bool) CSRF {
 }
 
 // Generate implements the CSRF interface
-func (c *cookieCsrf) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
+func (c *cookieCsrf) Generate(w http.ResponseWriter, req *http.Request) string {
 	cookie, err := req.Cookie(c.name)
 	if err == nil && len(cookie.Value) > 0 {
-		return cookie.Value, nil
+		return cookie.Value
 	}
 
 	cookie = &http.Cookie{
 		Name:     c.name,
-		Value:    uuid.NewUUID().String(),
```

Comment:
The old code was not random :/

Reply:
wait UUID is not random ?

Reply:
As far as I can tell is random

Reply:
Seems like it is; I missed it the first time I looked - not the best code organization imo: https://github.com/pborman/uuid/blob/master/uuid.go#L49

Reply:
`uuid.NewUUID.String()` is purely time based whereas `uuid.NewRandom().String()` and `uuid.New()` are actually random.

Reply:
I removed the direct dependency on the UUID package from our code entirely (hence the import restrictions change).  We keep trying to use it for random data, and clearly sometimes we do it incorrectly.